zstacklib/zstacklib/utils/file_metadata_handler.py

```
@@ -0,0 +1,206 @@
+                    # may have crashed before rename().  Do NOT promote it — the data
+                    # could be truncated.  Return None so the caller triggers a re-write.
+                    logger.warn("found orphan tmp file %s without final metadata; "
+                                "not promoting (may be incomplete)", tmp_path)
+                return {'metadata': None}
+
+            try:
+                with open(metadataPath, 'r') as f:
+                    content = f.read()
+                logger.debug("read vm metadata from %s (%d bytes)", metadataPath, len(content))
+                return {'metadata': content}
+            except Exception as e:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)